src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgery.cs

```diff
@@ -374,13 +374,28 @@ private IAntiforgeryFeature GetTokensInternal(HttpContext httpContext)
         /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
         protected virtual void SetDoNotCacheHeaders(HttpContext httpContext)
         {
-            // Since antifogery token generation is not very obvious to the end users (ex: MVC's form tag generates them
-            // by default), log a warning to let users know of the change in behavior to any cache headers they might
-            // have set explicitly.
-            LogCacheHeaderOverrideWarning(httpContext.Response);
+            bool cacheHeadersChanged = SetHeaderIfNotSet(httpContext, HeaderNames.CacheControl, "no-cache, no-store");
```

Comment:
I don't think this will fix the problem, the reason is the cookie is also a header, which is the main reason why it will keep failing.

What we need to do is generate the cookie/form antiforgery components and compare them to any existing one (if any) on the current request. If they are the same, we don't do anything. If they are different, we throw (as the identity of the user changed after generating tokens for the first time).